### PR TITLE
[Test] : add redis replication setup case

### PR DESCRIPTION
**Pull request by drivebyer**
**Description**

Related to : https://github.com/OT-CONTAINER-KIT/redis-operator/issues/656

**Type of change**

* Bug fix (non-breaking change which fixes an issue)
* New feature (non-breaking change which adds functionality)
* Breaking change (fix or feature that would cause existing functionality to not work as expected)

**Checklist**

- [ ] Tests have been added/modified and all tests pass.
- [ ] Functionality/bugs have been confirmed to be unchanged or fixed.
- [ ] I have performed a self-review of my own code.
- [ ] Documentation has been updated or added where necessary.

**Additional Context**

### CODE REVIEW

File: tests/e2e/v1beta2/setup/redis-replication/replication.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  name:  redis-replication
+spec:
+  clusterSize: 3
+  podSecurityContext:
+    runAsUser: 1000
+    fsGroup: 1000
+  # redisConfig:
+  #   additionalRedisConfig: redis-external-config
+  kubernetesConfig: 
+    image: quay.io/opstree/redis:v7.0.12
```
Comment (shubham-cmyk) [inline tests/e2e/v1beta2/setup/redis-replication/replication.yaml:14]: use `latest` tag here.

File: tests/e2e/v1beta2/setup/redis-replication/replication.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    imagePullPolicy: IfNotPresent
```
Comment (shubham-cmyk) [inline tests/e2e/v1beta2/setup/redis-replication/replication.yaml:15]:
```suggestion
    imagePullPolicy: Always
```

File: tests/e2e/v1beta2/setup/redis-replication/replication.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    image: quay.io/opstree/redis-exporter:v1.44.0
+    imagePullPolicy: Always
+    resources:
+      requests:
+        cpu: 100m
+        memory: 128Mi
+      limits:
+        cpu: 100m
+        memory: 128Mi
+# Environment Variables for Redis Exporter
```
Comment (shubham-cmyk) [inline tests/e2e/v1beta2/setup/redis-replication/replication.yaml:36]: Remove comments

File: tests/e2e/v1beta2/setup/redis-replication/replication.yaml (hunk 1 above)
Comment (shubham-cmyk) [reply tests/e2e/v1beta2/setup/redis-replication/replication.yaml:12]: no changed

Comment (drivebyer) [reply tests/e2e/v1beta2/setup/redis-replication/replication.yaml]: done

**Pull request merged**